### PR TITLE
Ruff toml begone (part 1 of many)

### PR DESCRIPTION
**Pull request by jelly**
Probably better to split this up in chunks because of test retries. Plus it fixes a CI spotted issue :)

### CODE REVIEW

**Review by martinpitt — APPROVED** [review]
Thank you! If that goes green, please land. We have some ENOSPC trouble on rhos-1. My train wifi isn't strong enough for doing VPN/ansible, until then please retry.

File: test/verify/check-shell-multi-machine [hunk 1]
```diff
@@ -451,7 +451,7 @@ class TestMultiMachine(MachineCase):
         m2 = self.machine2
 
         # Modify the terminals to be different on the two machines.
-        for machine, name in zip((m1, m2), ('m1', 'm2')):
+        for machine, name in zip((m1, m2), ('m1', 'm2'), strict=True):
```
Comment (allisonkarlitskaya) [inline test/verify/check-shell-multi-machine:454]: This requires a relatively new Python version....

Comment (martinpitt) [reply test/verify/check-shell-multi-machine:454]: This should be okay. This is a destructive test, i.e. it only runs in our Fedora 38 tasks container, not in downstream gating in e.g. CentOS 8.

Comment (jelly) [reply test/verify/check-shell-multi-machine:454]: Thanks for checking @martinpitt !

**Pull request merged**